src/test/scala/com/fortysevendeg/scala/android/unit/models/FormValidationSpec.scala

```diff
@@ -0,0 +1,135 @@
+package com.fortysevendeg.scala.android.unit.models
+
+import com.fortysevendeg.scala.android.modules.validators.DummyFormValidation
+import com.fortysevendeg.scala.android.ui.validateforms.FormModel
+import com.fortysevendeg.scala.android.{BaseTestSupport, WithUnitTestsSupport}
+import org.specs2.mutable.Specification
+
+class FormValidationSpec
+    extends Specification
+    with BaseTestSupport
+    with DummyFormValidation
+    with WithUnitTestsSupport {
+
+  "Name Validation in Dummy Form" should {
+
+    "return an error when the name field is empty" in {
+
+      val valueToTest = ""
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(Some(editableField), None, None)
+
+      val result = isValidNameField(form)
+
+      result.isSuccess === false
+      result.toEither.left.get.details === s"name can not be empty."
+    }
+
+    "return success when the name field is specified" in {
+
+      val valueToTest = "John Doe"
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(Some(editableField), None, None)
+
+      val result = isValidNameField(form)
+
+      result.isSuccess === true
+    }
+  }
+
+  "Email Validation in Dummy Form" should {
+
+    "return an error when the email field is empty" in {
+
+      val valueToTest = ""
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(None, Some(editableField), None)
+
+      val result = isValidEmailField(form)
+
+      result.isSuccess === false
+      result.toEither.left.get.details === s"email can not be empty."
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/47Degrees/scala-android/file/661377200/issues/source?bid=2339068&fileBranchId=2339108#l54)